Booting sequence stops at transpiling

Thanks to [vinnizworld feedback on reddit](https://www.reddit.com/r/effect/comments/1r0v2nw/comment/ob92zih/).

The boot sequence now stuck at:

```
boot
Boot effect starting...
boot
0/6 Boot started (scope entered)
boot
1/6 Acquiring semaphore...
boot
1/6 Semaphore acquired
boot
2/6 Booting WebContainer...
boot
2/6 WebContainer booted
boot
3/6 Fetching runtime.js...
boot
3/6 runtime.js fetched {"length":7248}
boot
3b/6 Initializing esbuild-wasm...
boot
3b/6 Transpiling initial program.js...
```

No changes, nor deployments have been made since 2 weeks ago, it was fine.

The chrome devtools output is :

```
index-DMUkEe7_.js:1258 [useWebContainerBoot] Boot started
8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:614 ✘ [ERROR] Cannot start service: Host version "0.27.3" does not match binary version "0.27.4"

8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:614 1 error
8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:188 exit code: 1
exit @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:188
runtime.wasmExit @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:310
$func955 @ esbuild.wasm:0xe138d
$func1404 @ esbuild.wasm:0x123c73
$func1288 @ esbuild.wasm:0x118ac0
$func2027 @ esbuild.wasm:0x1c35fc
$func5055 @ esbuild.wasm:0x91bc55
$func700 @ esbuild.wasm:0x9f3da
$func1395 @ esbuild.wasm:0x123b69
$run @ esbuild.wasm:0x123c4e
(anonymous) @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:577
(anonymous) @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:22
__async @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:6
run @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:517
(anonymous) @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:654
Promise.then
onmessage @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:651
(anonymous) @ 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:678
headless?version=1.6.1:1 The resource https://w-corp-staticblitz.com/fetch.worker.4c73681d.js was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
```

The important error is:

> 8ff6cd50-2f32-4b12-a1f8-c524b87b69a1:614 ✘ [ERROR] Cannot start service: Host version "0.27.3" does not match binary version "0.27.4"

There is a network request to `https://unpkg.com/esbuild-wasm@0.27.4/esbuild.wasm`

According to the error, it's a version mismatch error. 

Todo:

- Investigate then fix

--- a/.gitignore
+++ b/.gitignore
@@ -11,6 +11,7 @@ node_modules
 dist
 dist-ssr
 public/app
+public/esbuild.wasm
 *.local
 .local
 

--- a/package.json
+++ b/package.json
@@ -24,9 +24,10 @@
     "formatting"
   ],
   "scripts": {
-    "dev": "npm run build:runtime && vite",
-    "build": "npm run build:runtime && tsc -b && vite build",
+    "dev": "npm run copy:esbuild-wasm && npm run build:runtime && vite",
+    "build": "npm run copy:esbuild-wasm && npm run build:runtime && tsc -b && vite build",
     "build:runtime": "vite build --config vite.runtime.config.ts && tsc -p tsconfig.runtime.json && node scripts/extract-dts-from-runtime.mjs --src=dist/runtime > src/effects/app-lib-dts-urls.json && node scripts/copy-runtime.mjs",
+    "copy:esbuild-wasm": "cp node_modules/esbuild-wasm/esbuild.wasm public/esbuild.wasm",
     "lint": "eslint .",
     "lint:fix": "eslint . --fix",
     "preview": "vite preview",

--- a/src/lib/transpileForContainer.ts
+++ b/src/lib/transpileForContainer.ts
@@ -12,7 +12,9 @@ let initPromise: Promise<void> | null = null;
 export async function initEsbuildWasm(): Promise<void> {
   if (initPromise) return initPromise;
   initPromise = esbuild.initialize({
-    wasmURL: "https://unpkg.com/esbuild-wasm/esbuild.wasm",
+    // This file is copied to public/esbuild.wasm by the build script
+    // to ensure we work with the exact same version
+    wasmURL: "/esbuild.wasm",
   });
   return initPromise;
 }

